### ISSUE TITLE
Improperly formatted data in buildsystem_files.json

### ISSUE THREAD

**Issue by michaelbadcrumble** [mesonbuild/meson#6895]
**Describe the bug**
I detected an issue with Meson build files information not being formatted properly for Windows host via unit test in Meson-UI. Added a link to the relevant GitHub action https://github.com/michaelbadcrumble/meson-ui/runs/555201932?check_suite_focus=true

**To Reproduce**
N/A

**Expected behavior**
I expect OS appropriate path values when I choose to use `--buildsystem-files` or load the file from Meson-UI build directory API reader.

**system parameters**
* This is a CI build 
* Windows 10 and GitHub action Windows 10 image
* python 3.8.2
* meson 0.54.0
* ninja 1.10.0

**Comment by mensinda** [comment]
I don't think that this is necessarily a bug (Windows also accepts `/` as a path separator). However, there are some inconsistencies in other places where the host-specific path separator is used, if I read the introspection code correctly.

So, I am planning to make the output consistent by always using POSIX style paths. This is also consistent with meson using `/` as a path separator in the `meson.build` files. Is there a significant drawback for doing so on Windows?

**Comment by jpakkane** [comment]
AFAICR some tools intepret `/` as a command argument even if it is in the middle of a string. So this:

    foo bar/baz

is actually interpreted as

    foo bar /baz

Leading to an error about an unknown argument `/baz`. This may even come up with DOS standard commands like `copy`. The question is whether this is a big enough issue in practice.

**Comment by mensinda** [comment]
There is also the issue that you have to be really careful with passing strings that contain `\` (shell quoting, etc.). "Even" in CMake, the `\` has to be escaped.

So I would argue that everything in Windows *should* understand POSIX paths (at least in Powershell I hope) and it might improve the user experience (with all the shell quoting nonsense) if the same introspection scripts are used in Windows and Linux.

**Comment by michaelbadcrumble** [comment]
@mensinda and update on the issue?

**Comment by mensinda** [comment]
It is still on my todo list and my plan is to get the refactoring done (replace all `os.path.*` with `pathlib`) for `0.55.0`. Regarding #6921: At least `pathlib` works just fine with POSIX paths. Also, it doesn't help that POSIX and Windows path separators are used at the same time.

**Comment by bonzini** [comment]
Based on #6921 this was reproducible with https://github.com/dreamer-coding/meson-ui; if it is still possible to reproduce the issue please provide a recipe, so that it's possible to analyze whether the bug is in meson or something else.

However, my offhand guess is: introspection JSON uses forward slashes in paths consistently via `.as_posix()`, and if any programs want to convert them to backslashes they're free to do so.
